js/render/draw_fill.js

```diff
@@ -49,6 +50,39 @@ function draw(painter, source, layer, coords) {
         for (var j = 0; j < coords.length; j++) {
             drawStroke(painter, source, layer, coords[j]);
         }
+    } else if (layer.paint['fill-pattern']) {
```

Comment:
is this the correct conditional to have here? I was thinking ` else if (layer.paint['fill-pattern'] && !strokeColor)` might make sense? do we even need another conditional at all or would simply `else` suffice? I'm not super familiar with the possible permutations of the features checked in L27.